test/specs/rules/require-arrow-functions.js

```diff
@@ -0,0 +1,44 @@
+var Checker = require('../../../lib/checker');
+var assert = require('assert');
+
+describe('rules/require-arrow-functions', function() {
+    var checker;
+
+    beforeEach(function() {
+        checker = new Checker();
+        checker.registerDefaultRules();
+        checker.configure({ requireArrowFunctions: true, esnext: true });
+    });
+
+    it('should report use of function expression in declaration', function() {
+        assert(checker.checkString([
+            'var a = function(n) {',
+                'return n + 1;',
+            '});'
+        ].join('\n')).getErrorCount() === 1);
+    });
+
+    it('should report use of function expression as callback', function() {
+        assert(checker.checkString([
+            'a.map(function(n) {',
+                'return n + 1;',
+            '});'
+        ].join('\n')).getErrorCount() === 1);
+    });
+
+    it('should not report a function declaration', function() {
```

Comment:
Can we also add a test for named function expressions?

e.g., 

should still report on 

```js
var foo = function mySweetFunc() {};
```